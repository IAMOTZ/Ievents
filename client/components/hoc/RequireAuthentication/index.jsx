import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

@connect(store => ({
  isUserAuthenticated: store.authReducer.loggingUserResolved,
}))
class RequireAuthentication extends React.Component {
  render() {
    if (this.props.isUserAuthenticated) {
      return this.props.children;
    }
    return <Redirect to="/users/login" />;
  }
}

RequireAuthentication.propTypes = {
  isUserAuthenticated: PropTypes.bool,
  children: PropTypes.node,
};

RequireAuthentication.defaultProps = {
  isUserAuthenticated: false,
  children: null,
};

export default RequireAuthentication;
