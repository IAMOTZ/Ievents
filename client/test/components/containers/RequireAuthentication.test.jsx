import React from 'react';
import RequireAuthentication from '../../../components/containers/RequireAuthentication';

describe('<RequireAuthentication />', () => {
  const props = {
    isUserAuthenticated: null,
  };
  const childComponent = <p>You need authentication to see me</p>;
  describe('Rendering:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<RequireAuthentication {...props}>{childComponent}</RequireAuthentication>);
    });
    it('should redirect to login page if user is not authenticated', () => {
      wrapper.setProps({ isUserAuthenticated: false });
      expect(wrapper).toMatchSnapshot();
    });
    it('should render the child componetn is the user is authenticated', () => {
      wrapper.setProps({ isUserAuthenticated: true });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
