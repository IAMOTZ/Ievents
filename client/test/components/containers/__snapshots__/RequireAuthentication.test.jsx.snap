// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequireAuthentication /> Rendering: should redirect to login page if user is not authenticated 1`] = `
<Redirect
  push={false}
  to="/users/login"
/>
`;

exports[`<RequireAuthentication /> Rendering: should render the child componetn is the user is authenticated 1`] = `
<p>
  You need authentication to see me
</p>
`;
